backend/api/v1/routers/migrations.py

```
@@ -0,0 +1,340 @@
+@router.get("/{migration_id}/status", response_model=MigrationStatusResponse)
+async def get_migration_status(
+    migration_id: str,
+    service: MigrationOrchestrationService = Depends(get_migration_service)
+) -> MigrationStatusResponse:
+    """Get detailed migration status"""
+    try:
+        status = await service.get_migration_status(migration_id)
+        return MigrationStatusResponse(**status)
+        
+    except MigrationNotFoundError as e:
+        raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)